### PR TITLE
Add Integration tests to be run in a remote cluster

### PR DESCRIPTION
**Pull request by hhorak**
The idea is to have an easy way to run tests in an existing cluster, that might be remote via `make test-openshift-remote-cluster` or directly via `test/run-openshift-remote-cluster`.

### CODE REVIEW

**Comment by hhorak** [comment]
[test-openshift]

**Comment by ppisar** [comment]
You are missing a new 5.26-mod_fcgid subdirectory. Please note that that one
is ony for RHEL-8.

**Comment by hhorak** [comment]
[test-openshift]

**Comment by hhorak** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
Openshift tests pass for me locally. Maybe its just some timeout issue on centos?

**Comment by hhorak** [comment]
[test-openshift]

**Comment by hhorak** [comment]
Rebased, hope there is nothing missing now.

**Comment by hhorak** [comment]
[test-openshift]

File: test/run-openshift-local-cluster [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                        perl \
+                        "Everything is OK" \
+                        8080 http 200 "-p SOURCE_REPOSITORY_REF=staging -p VERSION=${VERSION} -p NAME=perl-testing"
+
+# test with the just built image and an integrated template
+test_perl_integration "${IMAGE_NAME}" "${VERSION}" perl
+
+# test with a released image and an integrated template
+if [ "${OS}" == "rhel7" ] ; then
+  PUBLIC_IMAGE_NAME=${PUBLIC_IMAGE_NAME:-${REGISTRY:-registry.access.redhat.com/}rhscl/${BASE_IMAGE_NAME}-${VERSION//./}-rhel7}
```
Comment (phracek) [inline test/run-openshift-local-cluster]: Please replace it with `registry.redhat.io`. Similar like here https://github.com/sclorg/postgresql-container/blob/master/test/run-openshift-local-cluster#L437

Look at whole section started from here https://github.com/sclorg/postgresql-container/blob/master/test/run-openshift-local-cluster#L435 to https://github.com/sclorg/postgresql-container/blob/master/test/run-openshift-local-cluster#L443

File: test/run-openshift-local-cluster [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                        ${THISDIR}/sample-test-app.json \
+                        perl \
+                        "Everything is OK" \
+                        8080 http 200 "-p SOURCE_REPOSITORY_REF=staging -p VERSION=${VERSION} -p NAME=perl-testing"
+
+# test with the just built image and an integrated template
+test_perl_integration "${IMAGE_NAME}" "${VERSION}" perl
+
+# test with a released image and an integrated template
+if [ "${OS}" == "rhel7" ] ; then
```
Comment (phracek) [inline test/run-openshift-local-cluster]:
```suggestion
# ignore possible failure of this test for centos images
fail_not_released=true
if [ "${OS}" == "rhel7" ] ; then
```

File: test/run-openshift-local-cluster [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  PUBLIC_IMAGE_NAME=${PUBLIC_IMAGE_NAME:-${REGISTRY:-registry.access.redhat.com/}rhscl/${BASE_IMAGE_NAME}-${VERSION//./}-rhel7}
+else
+  PUBLIC_IMAGE_NAME=${PUBLIC_IMAGE_NAME:-${REGISTRY:-}centos/${BASE_IMAGE_NAME}-${VERSION//./}-centos7}
```
Comment (phracek) [inline test/run-openshift-local-cluster]:
```suggestion
  PUBLIC_IMAGE_NAME=${PUBLIC_IMAGE_NAME:-${REGISTRY:-}centos/${BASE_IMAGE_NAME}-${VERSION//./}-centos7}
  fail_not_released=false
```

File: test/run-openshift-local-cluster [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+fi
+
+export CT_SKIP_UPLOAD_IMAGE=true
+test_perl_integration perl "${VERSION}" "${PUBLIC_IMAGE_NAME}"
```
Comment (phracek) [inline test/run-openshift-local-cluster]:
```suggestion
test_perl_integration perl "${VERSION}" "${PUBLIC_IMAGE_NAME}"
if docker pull "${PUBLIC_IMAGE_NAME}"; then
  test_perl_integration perl "${VERSION}" "${PUBLIC_IMAGE_NAME}"
else
  echo "Warning: ${PUBLIC_IMAGE_NAME} could not be downloaded via 'docker'"
  ! $fail_not_released || false "ERROR: Failed to pull image"
fi
```

Comment (phracek) [inline test/run-openshift-local-cluster]:
```suggestion
if docker pull "${PUBLIC_IMAGE_NAME}"; then
  test_perl_integration perl "${VERSION}" "${PUBLIC_IMAGE_NAME}"
else
  echo "Warning: ${PUBLIC_IMAGE_NAME} could not be downloaded via 'docker'"
  ! $fail_not_released || false "ERROR: Failed to pull image"
fi
```

**Comment by phracek** [comment]
This PR is covered by #188, which runs the tests in a remote cluster. Closing this issue. Feel free to reopen it.

**Pull request closed without merging**